cmd/os/windows/zfsinstaller/zfsinstaller.cpp

```diff
@@ -895,3 +898,102 @@ uninstallRootDevice(char *inf)
 	return (failcode);
 }
 #endif
+
+DWORD
+DeleteOemInf(const char* inf_name)
+{
+	TCHAR sysDir[MAX_PATH];
+	DWORD error = ERROR_SUCCESS;
+
+	if (!GetSystemDirectory(sysDir, MAX_PATH))
+		return GetLastError();
+
+	std::wstring infDir = std::wstring(sysDir).substr(0, (int)(_tcslen(sysDir) - _tcslen(_T("system32"))));
+	infDir += _T("inf\\");
+
+	TCHAR inf_nameW[MAX_PATH] = { 0 };
+	swprintf_s(inf_nameW, MAX_PATH, _T("%S"), inf_name);
+
+	std::wstring		mFullFileName;
+	TCHAR full[MAX_PATH];
+	if (_tfullpath(full, inf_nameW, MAX_PATH))
+		mFullFileName = full;
+	else
+		mFullFileName = inf_nameW;
+
+	// Check for existence.
+	if ((_taccess(mFullFileName.c_str(), 0)) == -1)
+		return (errno);
+
+	std::wstring oemName = infDir;
+	oemName += _T("oem*");
+	oemName += _T(".inf");
+
+	HANDLE hFile;
+	WIN32_FIND_DATA findFileData;
+	TCHAR szDriverFileName[MAX_PATH] = DRIVER_SYS_FILE;
+
+	if (INVALID_HANDLE_VALUE == (hFile = FindFirstFile(oemName.c_str(), &findFileData)))
+		return ERROR_SUCCESS;
+
+	do
+	{
+		// open each of inf file and search for the .sys file
+		// under the section [SourceDisksFiles]
+		TCHAR szValue[MAX_PATH];
+		std::wstring oemFile;
+		oemName = infDir + findFileData.cFileName;
+
+		if (0 < GetPrivateProfileString(
+			_T("SourceDisksFiles"),	// section name
+			szDriverFileName,		// lpKeyName
+			NULL,
+			szValue,				// lpReturnedString
+			MAX_PATH,
+			oemName.c_str()))		// lpFileName
+		{
+			BOOL success = SetupUninstallOEMInf(findFileData.cFileName, SUOI_FORCEDELETE, NULL);
+			if (!success)
+			{
+				error = GetLastError();
+				std::wcout << "\nFailed to delete the driver package from Driver Store.oemfile:" << findFileData.cFileName << ",error:" << error << std::endl;
+			}
+			else
+				std::wcout << "\nSuccessfully deleted the driver package from Driver Store.oemfile:" << findFileData.cFileName << std::endl;
```

Comment:
Why do we need to continue search after success? Is it possible to have multiple INF file for a driver?